aten/src/ATen/native/cuda/LegacyDefinitions.cpp

```diff
@@ -80,11 +80,13 @@ Tensor & masked_select_out_cuda(Tensor & result, const Tensor & self, const Tens
 }
 
 Tensor & gather_out_cuda(Tensor & result, const Tensor & self, int64_t dim, const Tensor & index, bool sparse_grad) {
+  result.resize_(index.sizes());
```

Comment:
nit: is there any type-check necessary here (like the one we discussed for the cummax implementation)?

Reply:
After offline chat. This is not needed as type checks are done in the THC binding!